frontend/packages/vkt/src/components/publicAuth/PublicAuthGrid.tsx

```diff
@@ -1,70 +1,138 @@
+
+  useEffect(() => {
+    if (selectedExamEvent && authStatus === APIResponseStatus.Success) {
+      dispatch(initialisePublicEnrollment(selectedExamEvent));
+    } else if (authStatus === APIResponseStatus.Error) {
+      navigate(AppRoutes.PublicHomePage, { replace: true });
+      dispatch(resetAuthentication());
+    }
+  }, [dispatch, navigate, selectedExamEvent, authStatus]);
+
+  useEffect(() => {
+    if (
+      selectedExamEvent &&
+      reservationDetailsStatus === APIResponseStatus.Success
+    ) {
       navigate(AppRoutes.PublicEnrollment, { replace: true });
+      dispatch(resetAuthentication());
     }
-  });
+  }, [navigate, dispatch, selectedExamEvent, reservationDetailsStatus]);
+
+  useEffect(() => {
+    if (
+      !ticket &&
+      authStatus === APIResponseStatus.NotStarted &&
+      reservationDetailsStatus === APIResponseStatus.NotStarted &&
+      !selectedExamEvent
+    ) {
+      navigate(AppRoutes.PublicHomePage, { replace: true });
+    }
+  }, [
+    navigate,
+    selectedExamEvent,
+    authStatus,
+    reservationDetailsStatus,
+    ticket,
+  ]);
 
   if (!selectedExamEvent) {
     return null;
   }
 
   const isExpectedToHaveOpenings = selectedExamEvent.openings > 0;
 
+  // TODO: change service query param to http%3A%2F%2Flocalhost%3A4002%2Fvkt%2Ftunnistaudu for local testing
+  const authUrl =
+    'https://testiopintopolku.fi/cas-oppija/login?service=https%3A%2F%2Fvkt%2Etestiopintopolku%2Efi%2Fvkt%2Ftunnistaudu';
+
   const renderDesktopView = () => (
     <>
       <Grid className="public-enrollment__grid" item>
         <Paper elevation={3}>
-          <div className="public-enrollment__grid__form-container">
-            <PublicEnrollmentStepper
-              activeStep={PublicEnrollmentFormStep.Authenticate}
-              includePaymentStep={isExpectedToHaveOpenings}
-            />
-            <PublicEnrollmentExamEventDetails
-              examEvent={selectedExamEvent}
-              showOpenings={isExpectedToHaveOpenings}
-            />
-            <div className="margin-top-xxl gapped rows">
-              <H3>{t('title')}</H3>
-              <CustomButton
-                // https://opintopolku.fi/cas-oppija/login?service=https://opintopolku.fi/JokuSovellus/SovelluksenLogin
-                href="https://testiopintopolku.fi/cas-oppija/login?service=http%3A%2F%2Flocalhost%3A4002%2Fvkt%2Ftunnistaudu"
-                sx={{ width: '168px' }}
-                variant={Variant.Contained}
-                color={Color.Secondary}
-                onClick={() => {
-                  //dispatch(initialisePublicEnrollment(selectedExamEvent));
-                }}
-                data-testid="public-enrollment__authenticate-button"
-                disabled={isLoading}
-              >
-                {t('buttonText')}
-              </CustomButton>
+          {reservationDetailsStatus === APIResponseStatus.InProgress ? (
+            <PublicAuthGridSkeleton />
+          ) : (
```

Comment:
Olisko selkeämpi jos tää tarkistus tehtäis jo PublicAuthPage komponentin alla?

Reply:
Aa, olin sattumalta tsekannu AKR:sta esimerkkiä PublicHomePage niin siellä itseasiassa oli tämä pyöräytetty Gridin alle niin valu sillä tänne. Page:ssa tosiaan suurin osa niin siirretään kans tonne tää